controllers/windowsmachine/windowsmachine_controller.go

```diff
@@ -116,10 +116,10 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		// We need the create event to account for Machines that are in provisioned state but were created
 		// before WMCO started running
 		CreateFunc: func(e event.CreateEvent) bool {
-			return isWindowsMachine(e.Meta.GetLabels())
+			return isValidMachine(e.Object.DeepCopyObject()) && isWindowsMachine(e.Meta.GetLabels())
```

Comment:
Is it safe to de-reference `e` without a check?

Reply:
@aravindhp I've changed this to not do a deep copy here. Could you elaborate if this what you meant?

Reply:
My question was if a `nil` check on `e` was required.

Reply:
I've added a nil check in the function :+1: